contracts/pcv/maker/MakerRouter.sol

```diff
@@ -0,0 +1,229 @@
+    )
+        external
+        hasAnyOfTwoRoles(TribeRoles.GOVERNOR, TribeRoles.PCV_CONTROLLER)
+    {
+        require(
+            ratioUSDC < Constants.BASIS_POINTS_GRANULARITY && ratioUSDC > 0,
+            "MakerRouter: Invalid USDC Ratio"
+        );
+
+        _redeemFromFeiPSM(amountFeiIn, minDaiAmountOut, address(this));
+        _swapForUsdcAndDai(minDaiAmountOut, usdcTo, daiTo, ratioUSDC);
+    }
+
+    /// @notice Function to swap all FEI balance for both DAI and USDC
+    /// @param usdcTo the address the USDC should be sent to once swapped
+    /// @param daiTo the address the DAI  should be sent to once swapped
+    /// @param ratioUSDC the ratio of the DAI received we would like to swap to USDC - in basis point terms
+    function swapAllFeiForUsdcAndDai(
+        address usdcTo,
+        address daiTo,
+        uint256 ratioUSDC
+    )
+        external
+        hasAnyOfTwoRoles(TribeRoles.GOVERNOR, TribeRoles.PCV_CONTROLLER)
+    {
+        uint256 minDaiAmountOut = _redeemAllBalanceFromFeiPSM(address(this));
+        _swapForUsdcAndDai(minDaiAmountOut, usdcTo, daiTo, ratioUSDC);
+    }
+
+    /// @notice Function to withdraw tokens to an address
+    /// @param token the token to withdraw
+    /// @param amount the amount to send
+    /// @param to the address the token should be sent to
+    function withdrawERC20(
+        address token,
+        uint256 amount,
+        address to
+    )
+        external
+        hasAnyOfTwoRoles(TribeRoles.GOVERNOR, TribeRoles.PCV_CONTROLLER)
+    {
+        IERC20(token).safeTransfer(to, amount);
+    }
+
+    /// @notice Helper function to redeem DAI from the FEI FEI-DAI PSM
+    /// @param amountFeiIn the amount of FEI to be deposited
+    /// @param minDaiAmountOut the minimum amount of DAI expected to be received
+    /// @param to the address the DAI should be sent to once swapped
+    function _redeemFromFeiPSM(
+        uint256 amountFeiIn,
+        uint256 minDaiAmountOut,
+        address to
+    ) private {
+        require(amountFeiIn > 1e18, "MakerRouter: Must deposit at least 1 FEI");
+        fei.safeTransferFrom(msg.sender, address(this), amountFeiIn);
+        feiPSM.redeem(to, amountFeiIn, minDaiAmountOut);
+    }
+
+    /// @notice Helper function to redeem DAI max allowance or balance from the FEI FEI-DAI PSM
+    /// @param to the address the DAI should be sent to once swapped
+    function _redeemAllBalanceFromFeiPSM(address to)
+        private
+        returns (uint256 minDaiAmountOut)
+    {
+        uint256 allowance = fei.allowance(msg.sender, address(this));
+        uint256 amountFeiIn = Math.min(fei.balanceOf(msg.sender), allowance);
+
+        require(amountFeiIn > 1e18, "MakerRouter: Must deposit at least 1 FEI");
+
+        fei.safeTransferFrom(msg.sender, address(this), amountFeiIn);
+        uint256 userBalanceBefore = dai.balanceOf(to);
+
+        // 3 is used as there is a 3 basis points redemption fee
+        minDaiAmountOut =
```

Comment:
You assume 1 FEI = 1 DAI, which will clearly lead to some slippage as both pegs fluctuate. Why not allow the user to input to the function a `minDAIAmountOut` so that they can control the slippage they get (and then apply the fee additionally inside the function)